File: cs12b/pa3/DictionaryTest.java
//DictionaryTest.java
//serve as a test client for the Dictionary ADT while it is under construction

public class DictionaryTest
{
	public static void main(String[] args)
	{
		Dictionary A = new Dictionary();
		String v;
		A.insert("1","a");
    	A.insert("2","b");
      	A.insert("3","c");
      	A.insert("4","d");
      	A.insert("5","e");
      	A.insert("6","f");
      	A.insert("7","g");
        //my test
        A.insert("20","g");
        A.insert("22","y");
        //A.insert("2","p");
        A.insert("42","i");
        A.insert("e","2");

      	System.out.println(A);

    	  v = A.lookup("1");
      	System.out.println("key=1 "+(v==null?"not found":("value="+v)));
      	v = A.lookup("3");
      	System.out.println("key=3 "+(v==null?"not found":("value="+v)));
      	v = A.lookup("7");
      	System.out.println("key=7 "+(v==null?"not found":("value="+v)));
      	v = A.lookup("8");
        System.out.println("key=8 "+(v==null?"not found":("value="+v)));
        //my test
        v = A.lookup("56");
        System.out.println("key=56 "+(v==null?"not found":("value="+v)));
        v = A.lookup("22");
        System.out.println("key=22 "+(v==null?"not found":("value="+v)));
        v = A.lookup("e");
        System.out.println("key=e "+(v==null?"not found":("value="+v)));
        v = A.lookup("20");
        System.out.println("key=20 "+(v==null?"not found":("value="+v)));
        v = A.lookup("0");
        System.out.println("key=0 "+(v==null?"not found":("value="+v)));

      	System.out.println();

      	//A.insert("2","f");  // causes DuplicateKeyException

      	A.delete("1");
      	A.delete("3");
      	A.delete("7");
        //my test
        A.delete("20");
        A.delete("e");
        //A.delete("3");
      	System.out.println(A);

      	//A.delete("8");  // causes KeyNotFoundException

     	System.out.println(A.isEmpty());
      	System.out.println(A.size());
      	A.makeEmpty();
      	System.out.println(A.isEmpty());
      	System.out.println(A);
		

      
	}
}

File: cs12b/pa5/DictionaryTest.c
// This is a Dictionary Test

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <assert.h>

#include "Dictionary.h"

#define MAX_LEN 180

int main(int argc,char **argv) {
   Dictionary A = newDictionary();
   char* word1[] = {"one","two","three","four","five","six","seven"};
   char* word2[] = {"foo","bar","blah","galumph","happy","sad","blue"};
   int i;

   for(i=0; i<7; i++) {
    insert(A, word1[i], word2[i]);
   }

   assert(size(A) == 7);
   assert(isEmpty(A) == 0);

   assert(strcmp(lookup(A,"one"),"foo") == 0);
   assert(strcmp(lookup(A,"two"),"bar") == 0);
   assert(strcmp(lookup(A,"three"),"blah") == 0);
   assert(strcmp(lookup(A,"four"),"galumph") == 0);
   assert(strcmp(lookup(A,"five"),"happy") == 0);
   assert(strcmp(lookup(A,"six"),"sad") == 0);
   assert(strcmp(lookup(A,"seven"),"blue") == 0);

   delete(A, "one");
   delete(A, "two");
   delete(A, "three");
   delete(A, "four");

   assert(lookup(A,"one") == NULL);
   assert(lookup(A,"two") == NULL);
   assert(lookup(A,"three") == NULL);
   assert(lookup(A,"four") == NULL);
   assert(strcmp(lookup(A,"five"),"happy") == 0);
   assert(strcmp(lookup(A,"six"),"sad") == 0);
   assert(strcmp(lookup(A,"seven"),"blue") == 0);

   assert(size(A) == 3);

   makeEmpty(A);
   assert(isEmpty(A) == 1);

   freeDictionary(&A);
   assert(A == NULL);
   return EXIT_SUCCESS;
}
